### PR TITLE
Deployment pipeline integration

### PR DESCRIPTION
**Pull request by ahmetalpbalkan**
Some housekeeping to integrate deployment of the new versions
through internal EDP tool.

cc: @boumenot

### CODE REVIEW

File: metadata/manifest.xml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+<?xml version='1.0' encoding='utf-8' ?>
+<!-- This XML file is used by Extension Deployment Pipeline (EDP) which
+     populates <MediaLink> and <Regions> elements appropriately -->
+<ExtensionImage xmlns="http://schemas.microsoft.com/windowsazure">
+  <!-- Caution: Ordering of elements in this document matters. -->
+  <ProviderNameSpace>Microsoft.Azure.Extensions</ProviderNameSpace>
+  <Type>DockerExtension</Type>
+  <Version>1.2.0</Version>
+  <Label>Microsoft Azure Custom Script Extension for Linux Virtual Machines</Label>
```
Comment (boumenot) [inline metadata/manifest.xml]: Oops!  Docker

**Comment by boumenot** [comment]
LGTM.  Missed one CSE to Docker replace.

File: metadata/manifest.xml (hunk 1 above)
Comment (ahmetalpbalkan) [inline metadata/manifest.xml]: ugh

**Comment by ahmetalpbalkan** [comment]
@boumenot ready to integrate!

Build: `make bundle`
Produces: `bundle/docker-extension.zip`
Integration tests: `echo '1.2.0' | ./integration-test/test.sh`

The problem though is these tests currently poll indefinitely when some test condition is not met inside the VM (e.g. a web container is running on port 80). It's already logged at #89. I guess that would cause a Jenkins job to keep running forever. (or we can run the tests with bash `timeout` command).

Second problem is the `integration-test/test.sh` does not get the extension namespace as parameter. So `*.Test` won't work just yet.

Perhaps for now, let's not integrate the tests to the EDP just yet.

**Comment by boumenot** [comment]
Sounds good.

It's easy to set a timeout in Jenkins.  This does mean you have to clean up....

**Comment by ahmetalpbalkan** [comment]
Hmm we have a way to cleanup things (basically just deleting vms with a certain prefix).

Integration tests require azure CLI (logged in and set in `asm` mode).

**Comment by boumenot** [comment]
This is exactly the methodology we use elsewhere.

We'll need to register the secret in Jenkins, but that's all fine.

**Pull request merged**